### PR TITLE
Feat/proper grpc init

### PR DESCRIPTION
**Pull request by kepkap**
1. proper grpc controllers registration
2. tollgate controller implementation added
3. minor refactoring

### CODE REVIEW

**Comment by vlkolmakov** [comment]
Please fix:
```go
func (r *r) Stop() error {
	if r.State.TryStop() {
		err := r.State.StopOrFail(r.PubSubClient.Close)
		if err != nil {
			return err
		}
		err = r.State.StopOrFail(r.StoreClient.Close)
		if err != nil {
			return err
		}
                return nil

	}
	return r.stateError()
}
```

File: internal/service/location/controller.go [hunk 1]
```diff
@@ -23,7 +27,18 @@ type Controller struct {
 	detector     *detector.Detector
 }
 
-func NewController(storeClient *redis.Client, pubSubClient *redis.Client) (*Controller, error) {
+func GrpcRegister() {
```
Comment (vlkolmakov) [inline internal/service/location/controller.go:30]: `init()`

File: main.go [hunk 2]
```diff
@@ -6,6 +6,11 @@ import (
 	"os"
 	"time"
 
+	"github.com/openmarketplaceengine/openmarketplaceengine/internal/service/location"
```
Comment (vlkolmakov) [inline main.go:9]: `_ "github.com/openmarketplaceengine/openmarketplaceengine/internal/service/location"`

File: main.go [hunk 3]
```diff
@@ -26,6 +31,11 @@ func init() {
 	boot.Add("STAT", stat.Boot, nil)
 	boot.Use("HTTP", srv.Http)
 	boot.Use("GRPC", srv.Grpc)
+
+	location.GrpcRegister()
+	tollgate.GrpcRegister()
+	crossing.GrpcRegister()
+	worker.GrpcRegister()
```
Comment (vlkolmakov) [inline main.go:38]: Remove

**Review by vlkolmakov — APPROVED** [review]

File: main.go (hunk 3 above)
Comment (kepkap) [reply main.go:38]: we agreed to migrate to init() and import package approach in another (next) PR.

**Pull request merged**